subworkflows/local/qc_workflow/test/main.nf.test

```diff
@@ -0,0 +1,129 @@
+                process {
+                    """
+                    input[0] = BEDOPS_CONVERT2BED.out.bed
+                    input[1] = Channel.of(
+                        [
+                            [id: 'chr22_dic'],
+                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/genome.dic", checkIfExist: true)
+                        ]
+                    )
+                    """
+                }
+            }
+
+        }
+
+        when {
+            // Params to activate modules ext.when condition
+            params {
+                skip_qc              = false
+                fusioninspector_only = false
+                starfusion           = true
+                all                  = true
+            }
+
+            workflow {
+                """
+                // ch_bam_sorted
+                input[0] = Channel.of(
+                    [
+                        [id: "chr22_bam"],
+                        file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam", checkIfExists: true)
+                    ])
+
+                // ch_chrgtf
+                input[1] = Channel.of(
+                    [
+                        [ id: "chr22_gtf" ],
+                        file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/genome.gtf", checkIfExists: true)
+                    ])
+
+                // ch_refflat
+                input[2] = UCSC_GTFTOGENEPRED.out.refflat
+
+                // ch_fasta
+                input[3] = Channel.of(
+                    [
+                        [ id: "test_ref" ],
+                        file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/genome.fasta", checkIfExist: true)
+                    ] )
+
+                // ch_fai
+                input[4] = Channel.of(
+                    [
+                        [ id: "test_ref" ],
+                        file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/genome.fasta.fai", checkIfExist: true)
+                    ] )
+
+                // ch rRNA interval list
+                input[5] = GATK4_BEDTOINTERVALLIST.out.interval_list
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success },
+                { assert snapshot(
+                    file( workflow.out.versions[0] ),
+                    file( workflow.out.rnaseq_metrics[0][1] ).readLines()[4..-1],
+                    file( workflow.out.duplicate_metrics[0][1] ).readLines()[4..-1],
+                    file( workflow.out.insertsize_metrics[0][1] ).readLines()[4..-1],
```

Comment:
I assume the metrics are non-stable, right? I think you can also get the md5 of that piece of stable content.

Reply:
Exactly, there's a date + time component there. Let me try again the md5 approach. I couldn't get it working yesterday